flood_adapt/object_model/hazard/measure/green_infrastructure.py

```diff
@@ -0,0 +1,106 @@
+import os
+from pathlib import Path
+from typing import Any, Union
+
+import geopandas as gpd
+import pyproj
+import tomli
+import tomli_w
+
+from flood_adapt.object_model.hazard.measure.hazard_measure import (
+    HazardMeasure,
+)
+from flood_adapt.object_model.interface.measures import (
+    GreenInfrastructureModel,
+    IGreenInfrastructure,
+)
+from flood_adapt.object_model.interface.site import ISite
+
+
+class GreenInfrastructure(HazardMeasure, IGreenInfrastructure):
+    """Subclass of HazardMeasure describing the measure of building a floodwall with a specific height"""
+
+    attrs: GreenInfrastructureModel
+    database_input_path: Union[str, os.PathLike]
+
+    @staticmethod
+    def load_file(filepath: Union[str, os.PathLike]) -> IGreenInfrastructure:
+        """create Floodwall from toml file"""
+
+        obj = GreenInfrastructure()
+        with open(filepath, mode="rb") as fp:
+            toml = tomli.load(fp)
+        obj.attrs = GreenInfrastructureModel.parse_obj(toml)
+        # if measure is created by path use that to get to the database path
+        obj.database_input_path = Path(filepath).parents[2]
+        return obj
+
+    @staticmethod
+    def load_dict(
+        data: dict[str, Any], database_input_path: Union[str, os.PathLike]
+    ) -> IGreenInfrastructure:
+        """create Green Infrastructure from object, e.g. when initialized from GUI"""
+
+        obj = GreenInfrastructure()
+        obj.attrs = GreenInfrastructureModel.parse_obj(data)
+        obj.database_input_path = database_input_path
+        return obj
+
+    def save(self, filepath: Union[str, os.PathLike]):
+        """save Floodwall to a toml file"""
+        with open(filepath, "wb") as f:
+            tomli_w.dump(self.attrs.dict(exclude_none=True), f)
+
+    @staticmethod
+    def calculate_volume(
+        area: float, height: float = 0.0, percent_area: float = 100.0
+    ) -> float:
+        """Determine volume from area of the polygon and infiltration height
+
+        Parameters
+        ----------
+        area : float
+            Area of polygon [m2] (calculated using calculate_polygon_area)
+        height : float, optional
+            Water height [m], by default 0.0
+        percent_area : float, optional
+            Percentage area covered by green infrastructure [%], by default 100.0
+
+        Returns
+        -------
+        float
+
+
+        Returns
+        -------
+        float
+            Volume [m3]
+        """
+        volume = area * height * percent_area / 100.0
```

Comment:
Why is there no unit conversion applied? Where is this function used? 

```suggestion
        volume = area.convert("m2") * height.convert("meters") * percent_area / 100.0
```